File: Leetcode/121-Best-Time-to-Buy-and-Sell-Stock/BestTimeToBuyAndSellStock.go
func maxProfit(prices []int) int {
	if prices == nil || len(prices) < 2 {
		return 0
	}
	profit := 0
	cost := prices[0]
	for _, value := range prices {
		profit = max(profit, value-cost)
		cost = min(cost, value)
	}
	return profit
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

File: Leetcode/121-Best-Time-to-Buy-and-Sell-Stock/BestTimeToBuyAndSellStock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices) <2:
            return 0;
        profit = 0
        cost = prices[0]
        for i in range(len(prices)):
            profit = max(profit, prices[i]-cost)
            cost = min(cost, prices[i])
        return profit
        
